Flaky apache_status integration test

Link: https://beats-ci.elastic.co/job/elastic+beats+master+multijob-darwin/beat=metricbeat,label=macosx/685/console


Test output:
```
20:57:53 command [go test -race -cover -coverprofile /var/folders/h7/35nznc4s255bn6_pkdm2fgdh0000kz/T/gotestcover-550085795 github.com/elastic/beats/metricbeat/module/apache/status]: exit status 1
20:57:53 --- FAIL: TestFetchTimeout (0.11s)
20:57:53 	assertions.go:247: 
20:57:53                           
20:57:53 	Error Trace:	status_test.go:170
20:57:53 		
20:57:53 	Error:      	An error is expected but got nil.
20:57:53 		
20:57:53 	Test:       	TestFetchTimeout
20:57:53 FAIL
20:57:53 coverage: 91.7% of statements
```

--- a/metricbeat/module/apache/status/status_test.go
+++ b/metricbeat/module/apache/status/status_test.go
@@ -147,13 +147,20 @@ func TestFetchEventContents(t *testing.T) {
 // TestFetchTimeout verifies that the HTTP request times out and an error is
 // returned.
 func TestFetchTimeout(t *testing.T) {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(200)
 		w.Header().Set("Content-Type", "text/plain; charset=ISO-8859-1")
 		w.Write([]byte(response))
-		time.Sleep(100 * time.Millisecond)
+
+		wg.Wait()
 	}))
-	defer server.Close()
+	defer func() {
+		wg.Done()
+		server.Close()
+	}()
 
 	config := map[string]interface{}{
 		"module":     "apache",

